Readme.md

```diff
@@ -528,6 +526,56 @@ cuba.addon.saml.logAllSamlMessages = true
 
 Also, you can observe the details of the implementation in the corresponding [demo project](https://git.haulmont.com/app-components/saml-addon-demo).
 
+## 4.2. Setup signing method for SAML messages <a name="setup-signing-method"></a>
+
+By default opensaml spring component uses sha1 digest algorithm for signing SAML messages. The most convenient way to use different signing message
+is create class in `WEB` module with addition changes in SecurityContext.
+
+<details><summary>Click to expand the example</summary>
+
+```java
+import org.opensaml.xml.Configuration;
+import org.opensaml.xml.security.BasicSecurityConfiguration;
+import org.opensaml.xml.signature.SignatureConstants;
+
+public class SecurityConfiguration {
+ 
+ public void initialize() {
+        BasicSecurityConfiguration configuration = (BasicSecurityConfiguration) Configuration.getGlobalSecurityConfiguration();
+
+        // Asymmetric key algorithms
+        configuration.registerSignatureAlgorithmURI("RSA", SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256);
+        configuration.registerSignatureAlgorithmURI("DSA", SignatureConstants.ALGO_ID_SIGNATURE_DSA);
+        configuration.registerSignatureAlgorithmURI("EC", SignatureConstants.ALGO_ID_SIGNATURE_ECDSA_SHA256);
+
+        // HMAC algorithms
+        configuration.registerSignatureAlgorithmURI("AES", SignatureConstants.ALGO_ID_MAC_HMAC_SHA256);
+        configuration.registerSignatureAlgorithmURI("DESede", SignatureConstants.ALGO_ID_MAC_HMAC_SHA256);
+
+        // Other signature-related params
+        configuration.setSignatureCanonicalizationAlgorithm(SignatureConstants.ALGO_ID_C14N_EXCL_OMIT_COMMENTS);
+        configuration.setSignatureHMACOutputLength(null);
+        configuration.setSignatureReferenceDigestMethod(SignatureConstants.ALGO_ID_DIGEST_SHA256);
+    }
+}
+```
+</details>
+
+
+Declare bean in `saml-dispatcher-spring.xml`
+
+```xml
+<!-- Initialization of OpenSAML library-->
+<bean id="samlBootstrap" class="org.springframework.security.saml.SAMLBootstrap"/>
+...
+<bean id="securityConfiguration" class="com.haulmont.addon.saml.web.SecurityConfiguration" init-method="initialize"
+          depends-on="samlBootstrap"/>
+```
+Basic configuration initilized in class `org.springframework.security.saml.SAMLBootstrap`. 
```

Comment:
Basic configuration **initialized** in **the** `org.springframework.security.saml.SAMLBootstrap` **class**.

Reply:
fixed